src/web/rustdoc.rs

```diff
@@ -223,20 +222,41 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         req_path.remove(0);
     }
 
-    version = match match_version(&conn, &name, url_version).and_then(|m| m.assume_exact()) {
-        Some(MatchSemver::Exact((v, _))) => v,
-        Some(MatchSemver::Semver((v, _))) => {
-            // to prevent cloudfront caching the wrong artifacts on URLs with loose semver
-            // versions, redirect the browser to the returned version instead of loading it
-            // immediately
-            let url = ctry!(Url::parse(
-                &format!("{}/{}/{}/{}", base, name, v, req_path.join("/"))[..]
-            ));
-            return Ok(super::redirect(url));
+    let redirect = |name: &str, vers: &str, path: &[&str]| -> IronResult<Response> {
+        let url = ctry!(Url::parse(
+            &format!(
+                "{}/{}/{}/{}",
+                redirect_base(req),
+                name,
+                vers,
+                path.join("/")
+            )[..]
+        ));
```

Comment:
This looks pretty ugly to me ... I assume that rustfmt wanted it this way? Can you try something like this?
```suggestion
        let redirect_path = format!("{}/{}/{}/{}", redirect_base(req), name, vers, path.join("/"));
        let url = ctry!(Url::parse(&redirect_path[..]));
```

(sorry for the continuing nits)

Reply:
Yes, this is a cargo fmt result, otherwise, we couldn't pass the `cargo fmt` step in CI. 😂

Reply:
Hmm I guess it's fine, it's not unreadable, just ugly.